avocado/utils/data_structures.py

```diff
@@ -33,6 +34,24 @@ def ordered_list_unique(object_list):
     return [x for x in object_list if not (x in seen or seen_add(x))]
 
 
+def geometric_mean(values):
+    """
+    Evaluates the geometric mean for a list of numeric values.
```

Comment:
PS: I'm not noting it to replace this implementation. The above is faster, but fails for bigger fields so here I think we're interested in slower, but stable results.